app/scripts/controllers/utils/utils.test.js

```diff
@@ -0,0 +1,92 @@
+import { strict as assert } from 'assert';
+import { checkSnapsBlockList } from './utils';
+
+describe('Controller utilities', function () {
+  describe('checkSnapsBlockList', function () {
+    it('returns one of the given snaps as blocked by its version', async function () {
+      const mockSnapsToBeChecked = {
+        'npm:exampleA': {
+          version: '1.0.0',
+          shasum: 'F5IapP6v1Bp7bl16NkCszfOhtVSZAm362X5zl7wgMhI=',
+        },
+        'npm:exampleB': {
+          version: '1.0.0',
+          shasum: 'eCYGZiYvZ3/uxkKI3npfl79kTQXS/5iD9ojsBS4A3rI=',
+        },
+        'npm:@consensys/starknet-snap': {
+          version: '0.1.10',
+          shasum: 'A83r5/ZIcKuKwuAnQHHByVFCuofj7jGK5hOStmHY6A0=',
+        },
+      };
+
+      const blockedSnaps = await checkSnapsBlockList(mockSnapsToBeChecked);
+      assert.deepEqual(blockedSnaps, {
+        'npm:exampleA': { blocked: false },
+        'npm:exampleB': { blocked: false },
+        'npm:@consensys/starknet-snap': {
+          blocked: true,
+          reason: undefined,
+          infoUrl: undefined,
+        },
+      });
+    });
+
+    it('returns given snap as blocked by its shasum', async function () {
+      const mockSnapsToBeChecked = {
+        'npm:@consensys/starknet-snap': {
+          version: '0.3.15', // try to fake version with the same source sha
```

Comment:
This now might be checked both by the version and the shasum right?
Using a mock `SNAP_BLOCKLIST` would ensure we only check one variant at a time.

Reply:
Yes, it now handles both (version and shasum). Updated the tests with mocked version of blocklist.